eth/beacon/state_machines/forks/serenity/operations.py

```diff
@@ -2,12 +2,51 @@
 from eth.beacon.types.pending_attestation_records import PendingAttestationRecord
 from eth.beacon.types.states import BeaconState
 from eth.beacon.state_machines.configs import BeaconConfig
+from eth.beacon.helpers import get_beacon_proposer_index
+from eth._utils.numeric import bitwise_xor
 
 from .validation import (
     validate_serenity_attestation,
+    validate_serenity_randao_reveal,
 )
 
 
+def validate_randao(state: BeaconState,
+                    block: BaseBeaconBlock,
+                    config: BeaconConfig) -> BeaconState:
+    """
+    Implements 'per-block-processing.RANDAO' portion of Phase 0 spec:
+    https://github.com/ethereum/eth2.0-specs/blob/master/specs/core/0_beacon-chain.md#randao
+
+    Verify the ``randao_reveal`` contained within the ``block`` in the context of
+    ``randao_commitment`` in the ``state``
+
+    """
+    proposer_index = get_beacon_proposer_index(state, state.slot, config.EPOCH_LENGTH)
+
+    validator_registry = list(state.validator_registry)
+    latest_randao_mixes = list(state.latest_randao_mixes)
+
+    proposer = validator_registry[proposer_index]
+
+    validate_serenity_randao_reveal(block, proposer)
+
+    latest_randao_mix = latest_randao_mixes[state.slot % config.LATEST_RANDAO_MIXES_LENGTH]
```

Comment:
We try to avoid overwriting variables.  Can you rename this first varaible to something like `original_randao_mix`